### ISSUE TITLE
appmetrics not working with mysql

### ISSUE THREAD

**Issue by chumberi** [RuntimeTools/appmetrics#458]
I am using appmetrics to monitor mysql changes and even though my application is performing mysql queries, I am not getting any data back.

`var appmetrics = require('appmetrics')
var monitoring = appmetrics.monitor();
monitoring.on("mysql", function (data) {
    console.log(data);
});`

I am using a db pool also, could this be an issue? Is there a workaround? Thanks

**Comment by tobespc** [comment]
@chumberi Which npm module are you using for your mysql queries ?

**Comment by chumberi** [comment]
@tobespc `const mysql = require('mysql');`

**Comment by mattcolegate** [comment]
@chumberi is your `require('appmetrics')` statement before your `require('mysql')` statement? If it's not, move your `require('appmetrics')` statement to the top of the file - Appmetrics instruments modules at require time, so it needs to be loaded first.

**Comment by chumberi** [comment]
@tobespc yes, unfortunately, my ` require('appmetrics')` is before my ` require('mysql') ` statement and its still not working...

**Comment by tobespc** [comment]
@chumberi any chance you could share the top half of your code where you are doing the requires ?

**Comment by chumberi** [comment]
@tobespc yup, no problem:

`var appmetrics = require('appmetrics')`
`var monitoring = appmetrics.monitor();`
`var metrics = ['http','http-outbound','cpu','gc','eventloop','loop','memcached','memory','mysql'];`
`const express = require('express');`
`const path = require('path');`
`const cookieParser = require('cookie-parser');`
`const bodyParser = require('body-parser');`
`const session = require('express-session');`
`const mysql = require('mysql');`

**Comment by tobespc** [comment]
@jamesemwallis Could you pick this up please and try and get a recreate scenario, thanks

**Comment by jamesemwallis** [comment]
@chumberi I have SQL data being emitted with the following simple test case.

[app.js.zip](https://github.com/RuntimeTools/appmetrics/files/1187274/app.js.zip)

Does this simple app work in your environment?
Could you share an example query that you are doing which should be getting data back as well please.

Thanks

**Comment by chumberi** [comment]
global.db_pool = mysql.createPool(config.dbDetails('write'));

query = **QUERY**;

global.db_pool.getConnection(function(err, connection) {
		connection.query(query, data, function(err, rows, fields) {
                })
});

**Comment by jamesemwallis** [comment]
I have recreated the problem and am looking into it now.

**Comment by jamesemwallis** [comment]
@chumberi I have been investigating the issue which is mysql pools. 

We do not currently support this but I will raise it as an enhancement request for the future.

Thanks.

**Comment by chumberi** [comment]
Has there been any update on this enhancement?

**Comment by sjanuary** [comment]
@chumberi we don't have any immediate plans to implement this but we are open to pull requests if you are interested in taking a look yourself.

**Comment by brenovieira** [comment]
Hey guys, any update or workaround on this enhancement?

**Comment by RedSpid3r** [comment]
Any update on this?
